bin/polykey.ts

```diff
@@ -0,0 +1,373 @@
+    .option('--verbose', 'increase verbosity by one level')
+    .action(async (options) => {
+        const password: string = options.password
+        const polykeyPath: string | undefined = options.polykeyPath
+        const publicKeyPath: string | undefined = options.publicKey
+        const privateKeyPath: string | undefined = options.privateKey
+        const privatePassphrase: string | undefined = options.privatePassphrase
+        const verbose: boolean | undefined = options.verbose
+        initPolyKey(
+            password,
+            polykeyPath,
+            publicKeyPath,
+            privateKeyPath,
+            privatePassphrase,
+            verbose
+        )
+    })
+
+
+
+/*******************************************/
+// keymanager create new keypair
+const keymanager = polykey.command('keymanager')
+    .description('manipulate the keymanager')
+
+/*******************************************/
+// node
+const node = polykey.command('node')
+node.description('network operations on the current polykey node')
+
+/*******************************************/
+// node start
+const nodeStart = node.command('start')
+    .description('start listening')
+    .action(async (options) => {
+        console.log('something something something');
+
+    })
+
+/*******************************************/
+// secrets
+const secrets = polykey.command('secrets')
+    .description('manipulate vault secrets')
+
+/*******************************************/
+// secrets list
+const secretsList = secrets.command('list')
+    .description('list all available secrets for a given vault')
+    .alias('ls')
+    .requiredOption('-v, --vault-name <vaultName>', 'the vault name')
+    .option('--verbose', 'increase verbosity level by one')
+    .action(async (options) => {
+        const isVerbose: boolean = options.verbose ?? false
+        const vaultName: string = options.vaultName
+        try {
+            // Check if vault exists
+            if (!(await pk.vaultExists(vaultName))) {
+                console.log(chalk.red(`vault '${vaultName}' does not exist!`))
+                return
+            }
+            // Get list of secrets from pk
+            const secretsList = await pk.listSecrets(vaultName)
+            // List secrets
+            if (isVerbose) {
+                console.log(`secrets contained within the ${vaultName} vault:`)
+            }
+            if (secretsList === undefined || secretsList.length == 0) {
+                console.log(`no secrets found for vault '${vaultName}'`)
+            } else {
+                secretsList.forEach((secretName) => {
+                    console.log(chalk.blueBright(secretName))
+                })
+            }
+        } catch (err) {
+            console.log(`Failed to list secrets for vault '${vaultName}': ${err.message}`)
+        }
+    })
+
+
+/*******************************************/
+// secrets add
+const secretAdd = secrets.command('add')
```

Comment:
This file is getting pretty long. Are we able to split up the commands so that way the entrypoint into the program can be a high level listing of the subcommands, and then each sub command can have their own file?

Reply:
Done.